### PR TITLE
🐛 (CLI; alpha generate) Preserve the customised Grafana config when alpha generate runs in place

### LINKED ISSUE #5966: The Grafana config does not survive an in place regeneration
We should check alpha generation with Grafana and ensure that we have good coverage.
The goal is to check: 

`migrateGrafanaPlugin` runs after the cleanup, and `grafanaConfigMigrate(src, des)` reads `<src>/grafana/custom-metrics/config.yaml`. For any in place run `src` and `des` are the same directory, so a customised Grafana dashboard config is deleted by the cleanup before it can be copied forward. The run logs `Grafana config file not found, skipping file migration` and the scaffold writes a default in its place.

I measured it on both refs, with a sentinel string in `grafana/custom-metrics/config.yaml` and the grafana plugin registered in `PROJECT`:

| case | `d9dafa9` (before #5920) | `85f3ae2f8` |
|---|---|---|
| plain `alpha generate` | customisation gone | customisation gone |
| `--input-dir` with no `--output-dir` | survives | customisation gone |

So the no flag case has been like this all along and is not something this PR introduces. `--input-dir` only starts sharing it because that directory is now correctly the one being cleaned. The old survival there was a side effect of scaffolding into the working directory, which is the bug this PR fixes, so I would not treat it as a reason to change anything here.

The fix, if you want one later, is already spelled out in the same function. `hack/boilerplate.go.txt` is read into memory before the cleanup and written back through a temp file afterwards. Reading the Grafana config the same way before cleaning would preserve it.

### PR DESCRIPTION
**Pull request by hexbinoct**
Fixes #5966

**What happens today**

`alpha generate` cleans the project directory before scaffolding. In an in place run (no flags, and since #5957 also `--input-dir` without `--output-dir`) the directory that gets cleaned is the same one `grafanaConfigMigrate(src, des)` later reads from. By that time the customised `grafana/custom-metrics/config.yaml` is gone, and the first `kubebuilder edit --plugins grafana...` has already scaffolded a default at that exact path, so the migration copies the default onto itself and the run ends with the customisation silently replaced.

**The fix**

Read the config before the cleanup and write that content back during the Grafana migration, the same way `hack/boilerplate.go.txt` is already preserved across the cleanup in this function. The `config.yaml` scaffold uses `SkipFile`, so the second `kubebuilder edit` pass does not overwrite the restored file, and the dashboard JSON (scaffolded with `OverwriteFile`) is regenerated from the restored config.

When `--output-dir` points to a separate directory the behaviour is unchanged: the preserved content is identical to the source file, which is still on disk.

**Verification**

Planted a sentinel string in `grafana/custom-metrics/config.yaml` of `testdata/project-v4` (layout set to v3, grafana plugin registered in PROJECT), then ran the migration on this branch and on master (829cce5) in golang:1.26 containers, each case twice:

| case | master | this branch |
|---|---|---|
| `alpha generate` in the project dir | replaced by the default | sentinel survives |
| `alpha generate --input-dir=<proj>` (no `--output-dir`) | replaced by the default | sentinel survives |

All runs exited 0. The unrelated working directory stayed untouched in the `--input-dir` case, and `.git` survived in every run. Unit tests: two new cases cover the in place restore and the preserved content taking precedence.

### CODE REVIEW

File: internal/cli/alpha/internal/generate.go [hunk 1]
```diff
@@ -751,13 +765,25 @@ func copyFile(src, des string) error {
...
 // Migrates Grafana configuration files.
-func grafanaConfigMigrate(src, des string) error {
+// preservedConfig holds the content of <src>/grafana/custom-metrics/config.yaml
+// as it was before the output directory was cleaned. When src and des are the
+// same directory (an in-place regeneration), the file on disk at this point is
+// a freshly scaffolded default, so the preserved content is the one to carry
+// forward.
+func grafanaConfigMigrate(src, des string, preservedConfig []byte) error {
+	desConfig := fmt.Sprintf("%s/grafana/custom-metrics/config.yaml", des)
+	if len(preservedConfig) > 0 {
```
Comment (sanmaxdev) [inline internal/cli/alpha/internal/generate.go]: `len(preservedConfig) > 0` treats an existing empty config the same as no config. During an in-place run, that leaves the scaffolded default in place instead of restoring the empty file. Please track existence separately, as the boilerplate path does, and add a regression case for an empty `config.yaml`.

**Review by sanmaxdev — COMMENTED** [review]
Focused migration tests pass. I found one preservation edge case below.

File: internal/cli/alpha/internal/generate.go (hunk 1 above)
Comment (hexbinoct) [reply internal/cli/alpha/internal/generate.go]: Good catch, thank you. An existing empty config.yaml took the copy path, and in place that source file is already the scaffolded default by migration time, so the default won. Fixed in a55beb2 by tracking existence in its own flag, the same way the boilerplate path does, and the restore now triggers on that flag. Added the regression case you asked for: an empty config.yaml is restored as an empty file. I also checked the new test fails against the previous check, so it does guard this.

File: internal/cli/alpha/internal/generate.go [hunk 2]
```diff
@@ -320,7 +337,7 @@ func migrateGrafanaPlugin(s store.Store, src, des string) error {
 		return fmt.Errorf("error editing Grafana plugin: %w", err)
 	}
 
-	if err = grafanaConfigMigrate(src, des); err != nil {
+	if err = grafanaConfigMigrate(src, des, preservedConfig, preservedConfigExists); err != nil {
```
Comment (camilamacedo86) [inline internal/cli/alpha/internal/generate.go:344]: When we do this call the files were deleted already.
I think the preservation in memory must to be before we delete and when we do it for PROJECT file.

Comment (hexbinoct) [reply internal/cli/alpha/internal/generate.go:344]: The preservation already happens before the delete, in the same block as the boilerplate one. In Generate() the config is read into memory right after the boilerplate read (lines 96 to 106), and the cleanup only runs later at line 121. Line 340 does not read from disk. It passes the bytes captured before the cleanup down to grafanaConfigMigrate, which writes them back once the new scaffold exists. The regression tests in generate_test.go cover the in place run for both a customised and an empty config.

**Review by camilamacedo86 — COMMENTED** [review]
HI @hexbinoct 

Really nice work 🥇 

For we get this one merged could you please just address the following nit ones:

  - Handle errors from os.Stat when checking the existing Grafana config. Only
    os.ErrNotExist should mean that the file is absent; permission or other
    filesystem errors should be returned before cleanup.

  - Add end-to-end regression coverage through Generate() for both in-place
    generation and --input-dir without --output-dir. The current tests call
    migrateGrafanaPlugin directly and do not verify that the configuration is
    captured before the project directory is cleaned.

**Comment by hexbinoct** [comment]
Thanks for the review! Both addressed.

**Stat errors**: the existence check now treats only `os.ErrNotExist` as "no config"; any other error from `os.Stat` is returned before the cleanup deletes anything, so a config that exists but cannot be read no longer gets silently lost.

**End to end coverage**: added two specs to the integration suite that #5969 merged, driving the real `alpha generate` command for both scenarios (in place, and `--input-dir` without `--output-dir`). Each scaffolds a project with the grafana plugin, customises `grafana/custom-metrics/config.yaml`, regenerates, and asserts the customisation survives and shows up in the regenerated dashboard. With the fix reverted both specs fail with the exact regression (the sentinel replaced by the scaffolded default), so they guard the capture before cleanup behaviour rather than the helper in isolation.

One note: I did not add a dedicated test for the stat error branch itself. It sits inline in Generate() and every portable simulation I could find is void somewhere (chmod 000 is a no-op as root, which is how the Prow containers run). Happy to add one if you have a preferred approach.

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

**Review by camilamacedo86 — APPROVED** [review]
/lgtm
